File: isma-compiler/hsm-core/src/main/java/ru/nstu/isma/compiler/hsm/core/linear/HMLinearVar.java
package ru.nstu.isma.compiler.hsm.core.linear;

import ru.nstu.isma.compiler.hsm.core.var.HMVariable;

import java.io.Serializable;

/**
 * Created by Bessonov Alex
 * on 13.03.2015.
 */
public class HMLinearVar extends HMVariable implements Serializable {
    private Integer columnIndex;

    public HMLinearVar() {
    }

    public HMLinearVar(String code) {
        super(code);
    }

    public Integer getColumnIndex() {
        return columnIndex;
    }

    public void setColumnIndex(Integer columnIndex) {
        this.columnIndex = columnIndex;
    }
}

File: isma-next-hsm/src/main/kotlin/ru/nstu/isma/hsm/linear/HMLinearVar.kt
package ru.nstu.isma.hsm.linear

import ru.nstu.isma.hsm.`var`.HMVariable
import java.io.Serializable

/**
 * Created by Bessonov Alex
 * on 13.03.2015.
 */
class HMLinearVar : HMVariable, Serializable {
    var columnIndex: Int? = null

    constructor() {}
    constructor(code: String?) : super(code) {}
}